pkg/trace/trace.go

```diff
@@ -35,14 +36,15 @@ func EnableOn(features []string) {
 }
 
 // On logs data to be shown if trace for a given feature is enabled and condition is met.
-func On(condition Condition, feature Feature, fields map[string]interface{}, format string, args ...interface{}) {
+func On(condition Condition, feature Feature, entry log.Entry, format string, args ...interface{}) {
 	if global.logger == nil {
 		return
 	}
 
 	if _, ok := global.enabled[feature]; ok && condition() {
-		if fields == nil {
-			fields = make(map[string]interface{})
+		fields := make(map[string]interface{})
+		if entry != nil {
+			fields = entry.Fields()
```

Comment:
I think Fields() could return a nil.

Reply:
thanks for the 👁️  @cristianciutea ! added changes in  https://github.com/newrelic/infrastructure-agent/pull/1070/commits/5912051a57439c7c21219e26e5989c4d61f02afc